src/org/zaproxy/zap/extension/history/PopupMenuExportURLs.java

```diff
@@ -79,7 +79,19 @@ public void actionPerformed(java.awt.event.ActionEvent e) {
         	    BufferedWriter fw = null;
                 try {
                     fw = new BufferedWriter(new FileWriter(file, isAppend));
-                    exportURLs((SiteNode) site.getModel().getRoot(), fw, html);
+                    int selectionCount=site.getSelectionCount();
+                    if (selectionCount == 0) {
+                        //No selection, export from root
+                        exportURLs((SiteNode) site.getModel().getRoot(), fw, html);
+                    } else if (selectionCount == 1) {

```

Comment:
The method `SiteMapPanel.getTreeContext()` needs to be public (or, add methods to `SiteMapPanel` to query/get the selection state of the contexts tree).